src/js/components/Notification/Notification.js

```diff
@@ -45,38 +94,74 @@ const Notification = ({ message, onClose, id, status, title, toast }) => {
   ]);
 
   const { icon: CloseIcon } = theme.notification.close;
-  const { icon: StatusIcon, color } = theme.notification[status];
+  const { icon: StatusIcon, background, color } = theme.notification[status];
   const { color: closeIconColor } = theme.notification.close;
 
+  const { direction, truncate } = toast
+    ? theme.notification.toast
+    : theme.notification;
+
+  const TextWrapper = direction === 'row' ? Text : Box;
+  const textWrapperProps = direction === 'row' ? { truncate } : {};
+
+  // notification is built with two child boxes that contain:
+  // 1. icon + text (wrapped in button when clickable)
+  // 2. close button
+  // pad needs to be applied to the child boxes to ensure pad is included
+  // in the clickable region, but we don't want to apply extra padding
+  // between the icon + text and the button.
+  const { pad } = theme.notification.container;
+  let textPad;
+  let closeButtonPad;
+  if (onClose) [textPad, closeButtonPad] = adaptThemeStyle(pad, theme);
+  else textPad = pad;
+
   let content = (
     <Box
       {...theme.notification.container}
       {...(toast ? { ...theme.notification.toast.container } : {})}
+      background={
+        !toast && background
+          ? background
+          : theme.notification.container.background
+      }
+      // let internal box control pad so clickable region includes pad
+      pad={undefined}
       direction="row"
+      gap={onClose ? 'small' : undefined}
     >
-      <Box {...theme.notification.iconContainer}>
-        <StatusIcon color={color} />
-      </Box>
-      <Box
-        {...theme.notification.textContainer}
-        align="start"
-        direction="row"
-        justify="between"
-        flex
-      >
-        <Box>
-          <Text {...theme.notification.title}>{title}</Text>
-          {message && (
-            <Paragraph {...theme.notification.message}>{message}</Paragraph>
+      {/* separate from onClose button to avoid nested interactive elements */}
+      <Wrapper href={href} onClick={onClick}>
+        <Box direction="row" pad={textPad} flex>
+          <Box {...theme.notification.iconContainer}>
+            <StatusIcon color={color} />
+          </Box>
+          <Box {...theme.notification.textContainer}>
+            <TextWrapper {...textWrapperProps}>
+              {title && <Text {...theme.notification.title}>{title}</Text>}
+              {message && title && direction === 'row' && <Text> </Text>}
+              {message && !truncate && direction !== 'row' ? (
+                <Paragraph {...theme.notification.message}>{message}</Paragraph>
+              ) : (
+                <Text {...theme.notification.message} truncate={truncate}>
+                  {message}
+                </Text>
+              )}
+            </TextWrapper>
+          </Box>
+        </Box>
+      </Wrapper>
+      <Box pad={closeButtonPad}>
```

Comment:
Can we remove this Box entirely if `!onClose`?

Reply:
Yes.